bcda/cclf/cclf.go

```diff
@@ -152,26 +149,55 @@ func importCCLF0(ctx context.Context, fileMetadata *cclfFileMetadata) (map[strin
 }
 
 func importCCLF8(ctx context.Context, fileMetadata *cclfFileMetadata) (err error) {
-	db := database.Connection
-
-	repository := postgres.NewRepository(db)
-
 	if fileMetadata == nil {
 		err = errors.New("CCLF file not found")
 		fmt.Println(err.Error())
 		log.Error(err)
 		return err
 	}
 
+	db := database.Connection
+	repository := postgres.NewRepository(db)
+
+	e, err := repository.GetCCLFFileExistsByName(ctx, fileMetadata.name)
+	if err != nil {
+		fmt.Printf("failed to check existence of CCLF%d file.\n", fileMetadata.cclfNum)
+		err = errors.Wrapf(err, "failed to check existence of CCLF%d file", fileMetadata.cclfNum)
+		log.Error(err)
+		return err
+	}
+
+	if e {
+		fmt.Printf("CCL%d file %s already exists in database, skipping import...", fileMetadata.cclfNum, fileMetadata)
+		log.Infof("CCL%d file %s already exists in database, skipping import...", fileMetadata.cclfNum, fileMetadata)
+		return nil
```

Comment:
Could we change the return type here (preferably a custom error) so that the caller of importCCLF8 knows if the file was already imported, other than through logs?

I don't think this needs to be done now, but maybe in a future ticket.